File: src/OeboWebApp/Models/Departure.cs
using System;

namespace OeboWebApp.Models
{
    public class Departure
    {
        public Crossing Crossing { get; set; }
        public DateTime Departs { get; set; }
        public DateTime Arrives { get; set; }
        public DepartureState State { get; set; }
    }

    public enum DepartureState
    {
        Available,
        SoldOut,
        Expired
    }
}

File: src/OeboWebApp/src/api/departure.ts
import { DateTime } from "luxon";
import { jsonProperty } from "ur-json";
import { luxonConverter } from "./converters/luxon-converter";
import { CrossingType } from "./crossing";

export type DepartureState = "available" | "soldOut" | "expired";

export class Departure {
    @jsonProperty()
    crossing!: CrossingType;
    @jsonProperty({ converter: luxonConverter })
    departs!: DateTime;
    @jsonProperty({ converter: luxonConverter })
    arrives!: DateTime;
    @jsonProperty()
    state!: DepartureState;
}